ast/src/analyzed/mod.rs

```diff
@@ -225,21 +227,29 @@ impl<T> Analyzed<T> {
 }
 
 #[derive(Debug, Clone)]
-pub struct Polynomial {
+pub struct Symbol {
     pub id: u64,
     pub source: SourceRef,
     pub absolute_name: String,
-    pub poly_type: PolynomialType,
+    pub kind: SymbolKind,
     pub degree: DegreeType,
     pub length: Option<DegreeType>,
 }
 
-impl Polynomial {
+impl Symbol {
     pub fn is_array(&self) -> bool {
         self.length.is_some()
     }
 }
 
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum SymbolKind {
+    /// Fixed, witness or intermediate polynomial
+    Poly(PolynomialType),
+    /// Other symbol, could be a constant, depends on the type.
+    Other(),
```

Comment:
As far as I can see, it's not currently instantiated ever, right? It's not very clear to me what this would be in the future.

Reply:
Haha true, let me try to "fix" that.